test/test_nn.py

```diff
@@ -68,7 +68,45 @@
 
 DOUBLE_TENSORTYPES = [torch.double]
 
-
+test_case1 = [
```

Comment:
please move these into the new class... especially when the variable names are this uninformative.